types/swap.go

```diff
@@ -0,0 +1,142 @@
+package types
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type SwapStatus byte
+
+const (
+	NULL      SwapStatus = 0x00
+	Open      SwapStatus = 0x01
+	Completed SwapStatus = 0x02
+	Expired   SwapStatus = 0x03
+)
+
+func NewSwapStatusFromString(str string) SwapStatus {
+	switch str {
+	case "Open", "open":
+		return Open
+	case "Completed", "completed":
+		return Completed
+	case "Expired", "expired":
+		return Expired
+	default:
+		return NULL
+	}
+}
+
+func (status SwapStatus) String() string {
+	switch status {
+	case Open:
+		return "Open"
+	case Completed:
+		return "Completed"
+	case Expired:
+		return "Expired"
+	default:
+		return "NULL"
+	}
+}
+
+func (status SwapStatus) MarshalJSON() ([]byte, error) {
+	return json.Marshal(status.String())
+}
+
+func (status *SwapStatus) UnmarshalJSON(data []byte) error {
+	var s string
+	err := json.Unmarshal(data, &s)
+	if err != nil {
+		return err
+	}
+	*status = NewSwapStatusFromString(s)
+	return nil
+}
+
+type SwapBytes []byte
+
+func (bz SwapBytes) Marshal() ([]byte, error) {
+	return bz, nil
+}
+
+func (bz *SwapBytes) Unmarshal(data []byte) error {
+	*bz = data
+	return nil
+}
+
+func (bz SwapBytes) MarshalJSON() ([]byte, error) {
```

Comment:
Can we use the built in Marshal/UmarshalJSON methods of the amino codec?

Reply:
https://github.com/Kava-Labs/go-sdk/pull/1/commits/b500f5d8773b86f005d2d6ea4ec750013e370060